test/evergreen.yml

```diff
@@ -5436,7 +5441,6 @@ buildvariants:
     # builds) as -O1, making test binaries more difficult to debug.
     CC_OPTIMIZE_LEVEL: -DCC_OPTIMIZE_LEVEL=-O0
     ENABLE_TCMALLOC: 1
```

Comment:
Is it expected to have tcmalloc for ubsan?

Reply:
It does not conflict with UBSan, but for simplicity I wonder if I shouldn't just remove it from **all** sanitizer tasks. Thoughts?

Reply:
Simplify by removing tcmalloc from all sanitizer tasks.